src/main/java/com/easypost/easyvcr/requestelements/Response.java

```diff
@@ -0,0 +1,434 @@
+                    headers[i] = new BasicHeader(s, matchingHeaderValues.get(i));
+                }
+                return headers;
+            }
+
+            @Override
+            public Header getFirstHeader(String s) {
+                Header[] headers = getHeaders(s);
+                if (headers == null || headers.length == 0) {
+                    return null;
+                }
+                return headers[0];
+            }
+
+            @Override
+            public Header getLastHeader(String s) {
+                Header[] headers = getHeaders(s);
+                if (headers == null || headers.length == 0) {
+                    return null;
+                }
+                return headers[headers.length - 1];
+            }
+
+            @Override
+            public Header[] getAllHeaders() {
+                Map<String, List<String>> headerMap = Response.this.getHeaders();
+                if (headerMap == null) {
+                    return null;
+                }
+                List<Header> headers = new ArrayList<>();
+                for (Map.Entry<String, List<String>> entry : headerMap.entrySet()) {
+                    for (String value : entry.getValue()) {
+                        headers.add(new BasicHeader(entry.getKey(), value));
+                    }
+                }
+                return headers.toArray(new Header[headers.size()]);
+            }
+
+            @Override
+            public void addHeader(Header header) {
+                // not implemented
+            }
+
+            @Override
+            public void addHeader(String s, String s1) {
+                // not implemented
+            }
+
+            @Override
+            public void setHeader(Header header) {
+                // not implemented
+            }
+
+            @Override
+            public void setHeader(String s, String s1) {
+                // not implemented
+            }
+
+            @Override
+            public void setHeaders(Header[] headers) {
+                // not implemented
+            }
+
+            @Override
+            public void removeHeader(Header header) {
+                // not implemented
+            }
+
+            @Override
+            public void removeHeaders(String s) {
+                // not implemented
+            }
+
+            @Override
+            public HeaderIterator headerIterator() {
+                // not implemented
+                return null;
+            }
+
+            @Override
+            public HeaderIterator headerIterator(String s) {
+                // not implemented
+                return null;
+            }
+
+            @Override
+            public HttpParams getParams() {
+                // not implemented
+                return null;
+            }
+
+            @Override
+            public void setParams(HttpParams httpParams) {
+                // not implemented
```

Comment:
For all these `not implemented` functions, do we want to throw a `not implemented` error instead?

Reply:
We can't throw any error, since that would change the function signature, which we can't do because it's an override.

Reply:
I was able to implement a few more, but for the most part, the user only received this object if it's a replayed request/response, and at this point in the workflow, the request was already made, so a user modifying these attributes after the fact doesn't make much sense in my mind anyway.